app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java

```diff
@@ -741,46 +734,58 @@ public void execute(Realm bgRealm) {
                                         rvChatFeed.getRecycledViewPool().clear();
                                         recyclerAdapter.notifyItemChanged((int) id);
                                         final String setMessage = answer;
-
+                                        final int counterValue = counter;
+                                        actionType = response.body().getAnswers().get(0).getActions().get(counter).getType();
                                         final Handler delayHandler = new Handler();
                                         delayHandler.postDelayed(new Runnable() {
                                             @Override
                                             public void run() {
-                                                addNewMessage(setMessage, ismap, isHavingLink, isPieChart, isSearchResult, datumList);
+                                                actionType = susiResponse.getAnswers().get(0).getActions().get(counterValue).getType();
+//                                                Log.e("Type:",actionType );
+
+                                                if ("anchor".equals(actionType)) {
+                                                    String text = susiResponse.getAnswers().get(0).getActions().get(counterValue).getAnchorText();
+                                                    String link = susiResponse.getAnswers().get(0).getActions().get(counterValue).getAnchorLink();
+                                                    addNewMessage(text.concat(": ".concat(link)), false, isHavingLink, false, false, datumList);
+                                                } else if ("answer".equals(actionType)) {
+                                                    addNewMessage(setMessage, ismap, isHavingLink, isPieChart, isSearchResult, datumList);
+//                                                } else if ("websearch".equals(actionType)) {
```

Comment:
@mayank408  For example, you want to fix the web search from here by uncommenting you can start to modify the contents.
It depends on which actionType we are modifying.